runtime/sema/access.go

```diff
@@ -191,12 +287,119 @@ func (e EntitlementMapAccess) IsLessPermissiveThan(other Access) bool {
 	}
 }
 
+func (e EntitlementMapAccess) Domain() EntitlementSetAccess {
+	if e.domain.Entitlements != nil {
+		return e.domain
+	}
+
+	domain := make(map[*EntitlementType]struct{})
+	for _, relation := range e.Type.Relations {
+		domain[relation.Input] = struct{}{}
+	}
+	e.domain = NewEntitlementSetAccess(maps.Keys(domain), Disjunction)
```

Comment:
Here and elsewhere: The result of [`maps.Keys`](https://pkg.go.dev/golang.org/x/exp/maps#Keys) (and `maps.Values`) has indeterminate order, it should not be used. Build up a slice yourself in the loop above, and maybe refactor the resulting code in a function and use it elsewhere.

We should probably enforce that `maps.Keys/Values` is not used in the codebase. All I can find is [depguard](https://github.com/OpenPeeDeeP/depguard), which unfortunately forbids whole package imports.